airbyte-integrations/connectors/destination-databricks/src/main/kotlin/io/airbyte/integrations/destination/databricks/sync/DatabricksStreamOperations.kt

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.databricks.sync
+
+import io.airbyte.cdk.integrations.destination.StreamSyncSummary
+import io.airbyte.cdk.integrations.destination.async.model.PartialAirbyteMessage
+import io.airbyte.cdk.integrations.destination.s3.FileUploadFormat
+import io.airbyte.commons.json.Jsons
+import io.airbyte.integrations.base.destination.typing_deduping.DestinationInitialStatus
+import io.airbyte.integrations.base.destination.typing_deduping.InitialRawTableStatus
+import io.airbyte.integrations.base.destination.typing_deduping.StreamConfig
+import io.airbyte.integrations.base.destination.typing_deduping.migrators.MinimumDestinationState
+import io.airbyte.integrations.destination.databricks.jdbc.DatabricksStorageOperations
+import io.airbyte.integrations.destination.databricks.staging.SerializableBufferFactory
+import io.airbyte.integrations.destination.sync.StreamOperations
+import io.airbyte.protocol.models.v0.DestinationSyncMode
+import io.github.oshai.kotlinlogging.KotlinLogging
+import java.util.*
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.atomic.AtomicReference
+import java.util.stream.Stream
+import org.apache.commons.io.FileUtils
+
+class DatabricksStreamOperations(
+    private val storageOperations: DatabricksStorageOperations,
+    private val fileUploadFormat: FileUploadFormat,
+) : StreamOperations<MinimumDestinationState.Impl> {
+    private val log = KotlinLogging.logger {}
+
+    // State maintained to make decision between async calls
+    private val initialRawTableStatus = AtomicReference<InitialRawTableStatus>(null)
```

Comment:
how do you feel about just having `private lateinit val initialRawTableStatus: InitialRawTableStatus`?

using an atomicreference here feels like a lot of extra ceremony for not much benefit? We can have SyncOperations validate that `destinationHandler.gatherInitialState` didn't return multiple values for the same stream, if we're worried about it

Reply:
yeah makes sense. I think originally i had it for additional state and it got stripped down and moved to Sync operations and forgot the atomicRef part.

Reply:
also findbugs was complaining i think, need to check again.